### ISSUE TITLE
npm install is throwing errors in Windows 10 Ubuntu Subsystem

### ISSUE THREAD

**Issue by bdbch** [npm/npm#12272]
Hey guys,

I had some troubling problems yesterday while trying out the new windows 10 build. I wanted to install some npm packages into my project but I couldn't make it after the npm-install task. I always get some EPERM-errors after the native dependencies should be installed. This happened in different projects and could be because of node-sass.

Maybe it's also an Ubuntu-related problem, since I don't use Ubuntu I can't tell this for 100%.

**Comment by c01nd01r** [comment]
+1
```
Windows 10 build 14316
npm 3.8.3
node 5.10.1

c01nd01r@localhost:~# sudo npm i gulp -g
Error: could not get uid/gid
net.js:135
    this._handle.open(options.fd);
                 ^

Error: EINVAL: invalid argument, uv_pipe_open
    at Error (native)
    at new Socket (net.js:135:18)
    at createWritableStdioStream (internal/process/stdio.js:142:16)
    at process.stdout (internal/process/stdio.js:10:14)
    at console.js:99:37
    at NativeModule.compile (node.js:395:5)
    at Function.NativeModule.require (node.js:340:18)
    at node.js:223:27
    at Object.<anonymous> (/usr/lib/node_modules/npm/node_modules/uid-number/get-uid-gid.js:16:1)
    at Module._compile (module.js:413:34)

    at /usr/lib/node_modules/npm/node_modules/uid-number/uid-number.js:37:16
    at ChildProcess.exithandler (child_process.js:209:5)
    at emitTwo (events.js:100:13)
    at ChildProcess.emit (events.js:185:7)
    at maybeClose (internal/child_process.js:850:16)
    at Socket.<anonymous> (internal/child_process.js:323:11)
    at emitOne (events.js:90:13)
    at Socket.emit (events.js:182:7)
    at Pipe._onclose (net.js:475:12)
```

**Comment by KISdesigners** [comment]
running on the same problem here

**Comment by JensVerneuer** [comment]
node sass is running for me but nodemon can not watch because of EPERM

**Comment by erwstout** [comment]
Same here, it looks as though dependencies aren't getting installed properly, here is a large strack trace when trying to run `gulp -T` for instance:

```sh
/mnt/d/git/ancora/wp-content/themes/ancora/node_modules/gulp-cssnano/node_modules/cssnano/node_modules/postcss-filter-plugins/node_modules/uniqid/node_modules/macaddress/index.js:31
    var ifaces = os.networkInterfaces();
                    ^

Error: EINVAL: invalid argument, uv_interface_addresses
    at Error (native)
    at Object.lib.networkInterfaces (/mnt/d/git/ancora/wp-content/themes/ancora/node_modules/gulp-cssnano/node_modules/cssnano/node_modules/postcss-filter-plugins/node_modules/uniqid/node_modules/macaddress/index.js:31:21)
    at Object.lib.one (/mnt/d/git/ancora/wp-content/themes/ancora/node_modules/gulp-cssnano/node_modules/cssnano/node_modules/postcss-filter-plugins/node_modules/uniqid/node_modules/macaddress/index.js:79:26)
    at Object.<anonymous> (/mnt/d/git/ancora/wp-content/themes/ancora/node_modules/gulp-cssnano/node_modules/cssnano/node_modules/postcss-filter-plugins/node_modules/uniqid/index.js:15:77)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
```

**Comment by LordPachelbel** [comment]
tl;dr - Netlink sockets are not supported in the version of WSL you're using but they were added to WSL as of Windows Insider build 14965 and will be available on stable in the Windows 10 Creator's Update this spring.

- https://github.com/Microsoft/BashOnWindows/issues/468
- https://github.com/scravy/node-macaddress/issues/8
- https://wpdev.uservoice.com/forums/266908-command-prompt-console-bash-on-ubuntu-on-windo/suggestions/16109719-enable-network-connection-enumeration
- https://blogs.msdn.microsoft.com/commandline/2016/11/10/ifconfig-and-network-connection-enumeration-support-in-windows-10-insider-build-14965/
